Is DCM-tools only available for 7.4?

I was hoping to use the DCM-tools to import the CA certificates provided by Mozilla to the *SYSTEM certificate store (the new QSYS2.HTTP_GET will not connect to https://www.ibm.com - certificate is not signed by trusted CA). So I installed the DCM-tools using the yum command as provided in the instructions.

But I get this error:
```
Transaction Check Error:
  package dcmtools-0.0.1-0.ppc64 is intended for a different operating system
```

And now I see the package is named "dcmtools-0.0.1-0._ibmi7.4_.ppc64" - so apparantly it is only build for IBM i 7.4. And I was trying to install it on a system with IBM i 7.3...

Will there be a version available for 7.3 as well?

I managed to download and install your version for 7.2 - but now I get another error:

```bash
→ dcmimport --ca-only --installed-certs
Successfully extracted installed certificates
Sanity check successful
Exception in thread "main" java.io.IOError: java.io.IOException: Qyj9SetEcho failed
        at java.io.Console.readPassword(Console.java:328)
        at com.github.ibmioss.dcmtools.utils.ConsoleUtils.askUserForPwd(ConsoleUtils.java:29)
        at com.github.ibmioss.dcmtools.DcmUserOpts.getDcmPassword(DcmUserOpts.java:30)
        at com.github.ibmioss.dcmtools.utils.DcmChangeTracker.<init>(DcmChangeTracker.java:92)
        at com.github.ibmioss.dcmtools.CertFileImporter.doImport(CertFileImporter.java:95)
        at com.github.ibmioss.dcmtools.DcmImportCmd.main(DcmImportCmd.java:128)
Caused by: java.io.IOException: Qyj9SetEcho failed
        at java.io.Console.echo(Native Method)
        at java.io.Console.readPassword(Console.java:326)
        ... 5 more
```

Another go - now after installing Java11 from ESS:

```
→ export JAVA_HOME=/QOpenSys/QIBM/ProdData/JavaVM/jdk11/64bit
→ dcmimport --installed-certs --ca-only
Successfully extracted installed certificates
Sanity check successful
Exception in thread "main" java.io.IOError: java.io.IOException: A specified file does not support the ioctl system call.
        at java.base/java.io.Console.readPassword(Console.java:318)
        at com.github.ibmioss.dcmtools.utils.ConsoleUtils.askUserForPwd(ConsoleUtils.java:29)
        at com.github.ibmioss.dcmtools.DcmUserOpts.getDcmPassword(DcmUserOpts.java:30)
        at com.github.ibmioss.dcmtools.utils.DcmChangeTracker.<init>(DcmChangeTracker.java:92)
        at com.github.ibmioss.dcmtools.CertFileImporter.doImport(CertFileImporter.java:95)
        at com.github.ibmioss.dcmtools.DcmImportCmd.main(DcmImportCmd.java:128)
Caused by: java.io.IOException: A specified file does not support the ioctl system call.
        at java.base/java.io.Console.echo(Native Method)
        at java.base/java.io.Console.readPassword(Console.java:316)
        ... 5 more
```

The java version is

```
→ java -version
openjdk version "11.0.11" 2021-04-20
OpenJDK Runtime Environment 11.0.11.0-IBM (build 11.0.11+9)
Eclipse OpenJ9 VM 11.0.11.0-IBM (build openj9-0.26.0, JRE 11 OS/400 ppc64-64-Bit Compressed References 20210427_000000 (JIT enabled, AOT enabled)
OpenJ9   - b4cc246
OMR      - 162e6f7
JCL      - 7796c80 based on jdk-11.0.11+9)
```

On Tue, Sep 28, 2021 at 1:23 PM Christian Jorgensen <
***@***.***> wrote:

>
> Caused by: java.io.IOException: A specified file does not support the ioctl system call.
>
>
Question: Is this in a 5250 session?


-- 
Jack Woehr, IBM Champion 2021
<https://www.youracclaim.com/badges/528d23d6-087f-4698-8d17-d59688106ac4/public_url>
Absolute Performance, Inc.
12303 Airport Way, Suite 100
Broomfield, CO 80021

NON-DISCLOSURE NOTICE:  This communication including any and all
attachments is for the intended recipient(s) only and may contain
confidential and privileged information.  If you are not the intended
recipient of this communication, any disclosure, copying further
distribution or use of this communication is prohibited.  If you received
this communication in error, please contact the sender and delete/destroy
all copies of this communication immediately.

--- a/README.md
+++ b/README.md
@@ -74,7 +74,7 @@ In any event, we're glad to have you aboard in any capacity, whether as a user, 
 # Installation
 
 ```
-yum install https://github.com/ThePrez/DCM-tools/releases/download/v0.0.1/dcmtools-0.0.1-0.ibmi7.4.ppc64.rpm
+yum install https://github.com/ThePrez/DCM-tools/releases/download/v0.1.0/dcmtools-0.1.0-0.ibmi7.2.ppc64.rpm
 ```
 Or, to build from source, clone this repository and run `make install`
 

--- a/README.md
+++ b/README.md
@@ -71,6 +71,9 @@ Feel free to [open an issue](https://github.com/ThePrez/DCM-tools/issues/new/cho
 
 In any event, we're glad to have you aboard in any capacity, whether as a user, spectator, or contributor!
 
+# IBM i release requirements
+The ability to "Renew" a certificate (with the `dcmrenew` tool) requires IBM i 7.4.
+
 # Installation
 
 ```

--- a/dcmtools.spec
+++ b/dcmtools.spec
@@ -1,10 +1,10 @@
 %undefine _disable_source_fetch
 Name: dcmtools
-Version: 0.0.1
+Version: 0.1.0
 Release: 0
 License: Apache-2.0
 Summary: Utilities for working with Digital Certificate Manager (DCM) for IBM i.
-Url: https://github.com/ThePrez/dcmimport
+Url: https://github.com/ThePrez/DCM-tools
 
 BuildRequires: maven
 BuildRequires: openjdk-11
@@ -36,5 +36,13 @@ INSTALL_ROOT=%{buildroot} gmake -e install
 %{_libdir}/%{name}
 
 %changelog
+* Sun Nov 14 2021 Jesse Gorzinski <jgorzins@us.ibm.com> - 0.1.0
+- feature: new 'dcmcreate' command
+- feature: new 'dcmview' command
+- feature: new 'dcmchangepw' command
+- feature: new 'dcmrenamecert' command
+- feature: new 'dcmremovecert' command
+- feature: provide binary install for IBM i 7.2
+
 * Wed Jul 28 2021 Jesse Gorzinski <jgorzins@us.ibm.com> - 0.0.1
 - initial RPM release

--- a/dcmtools.spec
+++ b/dcmtools.spec
@@ -43,6 +43,7 @@ INSTALL_ROOT=%{buildroot} gmake -e install
 - feature: new 'dcmrenamecert' command
 - feature: new 'dcmremovecert' command
 - feature: provide binary install for IBM i 7.2
+- bugfix: IBM i password is not hidden
 
 * Wed Jul 28 2021 Jesse Gorzinski <jgorzins@us.ibm.com> - 0.0.1
 - initial RPM release

--- a/scripts/dcm
+++ b/scripts/dcm
@@ -1,4 +1,9 @@
 #!/QOpenSys/pkgs/bin/bash
 export LC_ALL=EN_US.UTF-8
+if [[ -z "$SSH_TTY" ]]; then
+  unset TERM
+else
+  export TERM=xterm
+fi
 : ${JAVA_HOME=/QOpenSys/QIBM/ProdData/JavaVM/jdk80/64bit}
 exec $JAVA_HOME/bin/java -cp $(dirname $0)/../lib/dcmtools/dcmtools.jar com.github.ibmioss.dcmtools.DcmCmd $*

--- a/scripts/dcmassign
+++ b/scripts/dcmassign
@@ -1,4 +1,9 @@
 #!/QOpenSys/pkgs/bin/bash
 export LC_ALL=EN_US.UTF-8
+if [[ -z "$SSH_TTY" ]]; then
+  unset TERM
+else
+  export TERM=xterm
+fi
 : ${JAVA_HOME=/QOpenSys/QIBM/ProdData/JavaVM/jdk80/64bit}
 exec $JAVA_HOME/bin/java -cp $(dirname $0)/../lib/dcmtools/dcmtools.jar com.github.ibmioss.dcmtools.DcmAssignCmd $*

--- a/scripts/dcmchangepw
+++ b/scripts/dcmchangepw
@@ -1,4 +1,9 @@
 #!/QOpenSys/pkgs/bin/bash
 export LC_ALL=EN_US.UTF-8
+if [[ -z "$SSH_TTY" ]]; then
+  unset TERM
+else
+  export TERM=xterm
+fi
 : ${JAVA_HOME=/QOpenSys/QIBM/ProdData/JavaVM/jdk80/64bit}
 exec $JAVA_HOME/bin/java -cp $(dirname $0)/../lib/dcmtools/dcmtools.jar com.github.ibmioss.dcmtools.DcmChangePwCmd $*

--- a/scripts/dcmcreate
+++ b/scripts/dcmcreate
@@ -1,4 +1,9 @@
 #!/QOpenSys/pkgs/bin/bash
 export LC_ALL=EN_US.UTF-8
+if [[ -z "$SSH_TTY" ]]; then
+  unset TERM
+else
+  export TERM=xterm
+fi
 : ${JAVA_HOME=/QOpenSys/QIBM/ProdData/JavaVM/jdk80/64bit}
 exec $JAVA_HOME/bin/java -cp $(dirname $0)/../lib/dcmtools/dcmtools.jar com.github.ibmioss.dcmtools.DcmCreateCmd $*

--- a/scripts/dcmexport
+++ b/scripts/dcmexport
@@ -1,4 +1,9 @@
 #!/QOpenSys/pkgs/bin/bash
 export LC_ALL=EN_US.UTF-8
+if [[ -z "$SSH_TTY" ]]; then
+  unset TERM
+else
+  export TERM=xterm
+fi
 : ${JAVA_HOME=/QOpenSys/QIBM/ProdData/JavaVM/jdk80/64bit}
 exec $JAVA_HOME/bin/java -cp $(dirname $0)/../lib/dcmtools/dcmtools.jar com.github.ibmioss.dcmtools.DcmExportCmd $*

--- a/scripts/dcmexportcert
+++ b/scripts/dcmexportcert
@@ -1,4 +1,9 @@
 #!/QOpenSys/pkgs/bin/bash
 export LC_ALL=EN_US.UTF-8
+if [[ -z "$SSH_TTY" ]]; then
+  unset TERM
+else
+  export TERM=xterm
+fi
 : ${JAVA_HOME=/QOpenSys/QIBM/ProdData/JavaVM/jdk80/64bit}
 exec $JAVA_HOME/bin/java -cp $(dirname $0)/../lib/dcmtools/dcmtools.jar com.github.ibmioss.dcmtools.DcmExportCertCmd $*

--- a/scripts/dcmimport
+++ b/scripts/dcmimport
@@ -1,4 +1,9 @@
 #!/QOpenSys/pkgs/bin/bash
 export LC_ALL=EN_US.UTF-8
+if [[ -z "$SSH_TTY" ]]; then
+  unset TERM
+else
+  export TERM=xterm
+fi
 : ${JAVA_HOME=/QOpenSys/QIBM/ProdData/JavaVM/jdk80/64bit}
 exec $JAVA_HOME/bin/java -cp $(dirname $0)/../lib/dcmtools/dcmtools.jar com.github.ibmioss.dcmtools.DcmImportCmd $*

--- a/scripts/dcmremovecert
+++ b/scripts/dcmremovecert
@@ -1,4 +1,9 @@
 #!/QOpenSys/pkgs/bin/bash
 export LC_ALL=EN_US.UTF-8
+if [[ -z "$SSH_TTY" ]]; then
+  unset TERM
+else
+  export TERM=xterm
+fi
 : ${JAVA_HOME=/QOpenSys/QIBM/ProdData/JavaVM/jdk80/64bit}
 exec $JAVA_HOME/bin/java -cp $(dirname $0)/../lib/dcmtools/dcmtools.jar com.github.ibmioss.dcmtools.DcmRemoveCertCmd $*

--- a/scripts/dcmrenamecert
+++ b/scripts/dcmrenamecert
@@ -1,4 +1,9 @@
 #!/QOpenSys/pkgs/bin/bash
 export LC_ALL=EN_US.UTF-8
+if [[ -z "$SSH_TTY" ]]; then
+  unset TERM
+else
+  export TERM=xterm
+fi
 : ${JAVA_HOME=/QOpenSys/QIBM/ProdData/JavaVM/jdk80/64bit}
 exec $JAVA_HOME/bin/java -cp $(dirname $0)/../lib/dcmtools/dcmtools.jar com.github.ibmioss.dcmtools.DcmRenameCertCmd $*

--- a/scripts/dcmrenew
+++ b/scripts/dcmrenew
@@ -1,4 +1,9 @@
 #!/QOpenSys/pkgs/bin/bash
 export LC_ALL=EN_US.UTF-8
+if [[ -z "$SSH_TTY" ]]; then
+  unset TERM
+else
+  export TERM=xterm
+fi
 : ${JAVA_HOME=/QOpenSys/QIBM/ProdData/JavaVM/jdk80/64bit}
 exec $JAVA_HOME/bin/java -cp $(dirname $0)/../lib/dcmtools/dcmtools.jar com.github.ibmioss.dcmtools.DcmRenewCmd $*

--- a/scripts/dcmrenewacme
+++ b/scripts/dcmrenewacme
@@ -1,5 +1,10 @@
 #!/QOpenSys/pkgs/bin/bash
 exec echo "not implemented"
 export LC_ALL=EN_US.UTF-8
+if [[ -z "$SSH_TTY" ]]; then
+  unset TERM
+else
+  export TERM=xterm
+fi
 : ${JAVA_HOME=/QOpenSys/QIBM/ProdData/JavaVM/jdk80/64bit}
 exec $JAVA_HOME/bin/java -cp $(dirname $0)/../lib/dcmtools/dcmtools.jar com.github.ibmioss.dcmtools.DcmRenewAcmeCmd $*

--- a/scripts/dcmview
+++ b/scripts/dcmview
@@ -1,4 +1,9 @@
 #!/QOpenSys/pkgs/bin/bash
 export LC_ALL=EN_US.UTF-8
+if [[ -z "$SSH_TTY" ]]; then
+  unset TERM
+else
+  export TERM=xterm
+fi
 : ${JAVA_HOME=/QOpenSys/QIBM/ProdData/JavaVM/jdk80/64bit}
 exec $JAVA_HOME/bin/java -cp $(dirname $0)/../lib/dcmtools/dcmtools.jar com.github.ibmioss.dcmtools.DcmViewCmd $*

--- a/src/main/java/com/github/ibmioss/dcmtools/DcmViewCmd.java
+++ b/src/main/java/com/github/ibmioss/dcmtools/DcmViewCmd.java
@@ -64,7 +64,7 @@ public class DcmViewCmd {
 
     private static void printUsageAndExit() {
         // @formatter:off
-		final String usage = "Usage: dcmexport [options]\n"
+		final String usage = "Usage: dcmview [options]\n"
 		                        + "\n"
 		                        + "    Valid options include:\n"
                                 + "        -y:                            Do not ask for confirmation\n"

